javaslang/src/test/java/javaslang/collection/QuickSort.java

```diff
@@ -0,0 +1,25 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _______
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  //  /\__\   JΛVΛSLΛNG
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/ \ /__\ \   Copyright 2014-2017 Javaslang, http://javaslang.io
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/\__\_/  \_//  \__/\_____/   Licensed under the Apache License, Version 2.0
+ */
+package javaslang.collection;
+
+import org.junit.Test;
+
+import static javaslang.API.Vector;
+
+public class QuickSort {
+    @Test
+    public void shouldQuickSort() {
+        final Vector<Integer> values = Vector(3, 1, 4, 1, 5, 9, 2, 6, 5, 3, 5, 8, 9, 7, 9, 3, 2, 3, 8, 4, 6, 2, 6);
+        assert sort(values).equals(values.sorted());
+    }
+
+    /** Note: this example is only meant to show off, not to be used in reality: it can have quadratic performance and cause stack overflow */
+    private static Seq<Integer> sort(Seq<Integer> values) {
+        if (values.size() <= 1) return values;
+        else return values.tail().partition(v -> v <= values.head())
```

Comment:
@paplorinc Yes:

```java
return Match(values).of(
    Case(Patterns.List($(), $()), (x, xs) ->
            xs.partition(v -> v <= x)
              .apply((less, more) -> sort(less).append(values.head()).appendAll(sort(more)))),
    Case($(), () -> values)
);
```

Spoiler alert: We use javaslang.Patterns.List here, which will be obsolete in near future...